py3status/modules/xrandr.py

```diff
@@ -141,17 +156,28 @@ def _set_available_combinations(self):
         available_combinations = set()
         combinations_map = {}
 
+        if self.output_combinations is None:
+            whitelist = None
+        else:
+            whitelist = self.output_combinations.split('|')
+
         self.max_width = 0
         for output in range(len(self.layout['connected']) + 1):
             for comb in combinations(self.layout['connected'], output):
-                if comb:
-                    for mode in ['clone', 'extend']:
-                        string = self._get_string_and_set_width(comb, mode)
-                        if len(comb) == 1:
-                            combinations_map[string] = (comb, None)
-                        else:
-                            combinations_map[string] = (comb, mode)
-                        available_combinations.add(string)
+                if not comb:
+                    continue
+                for mode in ['clone', 'extend']:
+                    string = self._get_string_and_set_width(comb, mode)
+                    if whitelist is not None and string not in whitelist:
+                        continue
+                    combinations_map[string] = (comb, None if len(comb) == 1 else mode)
+                    available_combinations.add(string)
```

Comment:
Personally I find this harder to follow than the existing code.

```
if len(comb) == 1:
     combinations_map[string] = (comb, None)
else:
     combinations_map[string] = (comb, mode)
```

might be longer than

```
combinations_map[string] = (comb, None if len(comb) == 1 else mode)
```
but it is much easier to maintain and is really clear what it's doing. 

There is also code churn which makes reviews harder and so increases the chance of introducing bugs.  Don't get me wrong I'm a fan of re-factoring stuff but the aim should be to make it clearer to read.